doc/designs/fips_indicator.md

```diff
@@ -0,0 +1,262 @@
+
+- Ciphers
+
+``` C
+EVP_CIPHER_CTX_get_params()
+```
+
+- Digests
+
+``` C
+EVP_MD_CTX_get_params()
+```
+
+- KDF’s
+
+``` C
+EVP_KDF_CTX_get_params()
+```
+
+- MAC’s
+
+``` C
+EVP_MAC_CTX_get_params()
+```
+
+- RAND
+
+``` C
+EVP_RAND_CTX_get_params()
+```
+
+### Backwards Compatibility
+
+Previous providers do not support this operation, so they will return not approved if they are not handled.
+
+### Alternate Solution
+
+If we had different kinds of compliance requirements (something other than FIPS) either a separate getter could be added or the getter could return a int type instead of a 0 or 1..
+(e.g 1 = fips approved, 2 = some other compliance approved)
+
+Changes Required for indicators
+-------------------------------
+
+### key size >= 112 bits
+
+There are a few places where we do not enforce key size that need to be addressed.
+
+- HMAC  Which applies to all algorithms that use HMAC also.
+- CMAC
+
+### Algorithm Transitions
+
+Should we remove these algorithms completely from the fips provider, or use indicators?
+
+- DES_EDE3_ECB.  Disallowed for encryption, allowed for legacy decryption
+- DSA.  Keygen and Signing are no longer approved, not sure if verify is still approved.
+- RSA Signing using PKCSV15.  Rsa self test for sign may need to change as the default pad_mode is PKCSV15. Check if saltlen needs a indicator. Padding mode updates required in rsa_check_padding(). Check if sha1 is allowed?
+- ECDSA B & K curves are deprecated   It looks like these are still allowed. Are the approved?
+- ED25519/ED448 is now approved.
+- X25519/X448 is not approved currently. keygen would need an indicator if we allow it?
+- RSA Key transport  Padding need to be changed to either not allow padding modes other than OEAP or use an indicator.
+- RSA - It looks like SP800-131Ar2 specifies that RSA >= 2048 is approved. Can we check with the labs what to do for the legacy verification cases since we allow >=1024. Would this now be unapproved? This would apply to rsa_kem also? rsa_keygen_pairwise_test() may need to change to do a sign/verify PCT?
+- TLS1_PRF  If we are only trying to support TLS1.2 here then we should remove the tls1.0/1.1 code from the FIPS MODULE.
+
+### Digest Checks
+
+Any algorithms that use a digest need to make sure that the CAVP certificate lists all supported FIPS digests otherwise an indicator is required.
+This applies to the following algorithms:
+
+- SSKDF
+- TLS_1_3_KDF
+- SSHKDF
+- X963KDF
+- X942KDF
+- PBKDF2
+- HKDF
+- TLS1_PRF
+- HMAC
+- KBKDF
+- KMAC
+
+Note this includes SHAKE
+Do we need to check which algorithms allow SHA1 also?
+
+### Cipher Checks
+
+- CMAC
+- KBKDF CMAC
+- GMAC
+
+We should only allow AES. We currently just check the mode.
+
+### Configurable options
+
+- PBKDF2 'lower_bound_checks' needs to be part of the indicator check
+
+- See the "security checks" Section. Anywhere using ossl_securitycheck_enabled() may need an indicator
+
+Other Changes
```

Comment:
Thanks for the input...
I would be fine with K & B curves being removed altogether - but if we keep them then what you say applies.
I do mention salt length in the Algorithm Transitions - it will need to be checked. I think it needs an indicator.
- RSA OEAP digests should be determined by what we can actually test.
-Looks like KMAC needs an indicator on the lower bound.

Reply:
According to [the specification](https://github.com/usnistgov/ACVP/blob/master/src/kas/sp800-56br2/sections/05-capabilities.adoc) (Table 12), ACVP testing for OAEP is restricted to SHA-1, SHA-2, or SHA-3. Can also be confirmed by looking at the ACVP server [source code](https://github.com/usnistgov/ACVP-Server/blob/master/gen-val/src/crypto/src/NIST.CVP.ACVTS.Libraries.Crypto/KTS/KtsFactory.cs#L22)